myWumpus

```diff
@@ -5,16 +5,73 @@ We want to write a program that consists of
 and perform logic inference.
 
 """
+def to_cnf(sentence):
+    """
+    Convert a propositional logic sentence into CNF (Conjunctive Normal Form).
+    """
+    if isinstance(sentence, str):
+        return sentence  # atomic proposition
+
+    operator, *operands = sentence
+
+    if operator == 'NOT':
+        sub_cnf = to_cnf(operands[0])
+        if isinstance(sub_cnf, tuple) and sub_cnf[0] == 'NOT':
+            return sub_cnf[1]  # double negation elimination
+        return ('NOT', sub_cnf)
+
+    elif operator == 'AND':
+        return ('AND', to_cnf(operands[0]), to_cnf(operands[1]))
+
+    elif operator == 'OR':
+        return ('OR', to_cnf(operands[0]), to_cnf(operands[1]))
+
+    elif operator == 'IMPLIES':
+        # P => Q is equivalent to NOT P OR Q
+        left = to_cnf(operands[0])
+        right = to_cnf(operands[1])
+        return ('OR', ('NOT', left), right)
```

Comment:
Very nice. You are the first one I've seen who applied the recursive call on the individual clauses.